spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -659,6 +700,19 @@ protected Object handleRequestMessage(final Message<?> message) {
 				jmsReply = this.sendAndReceiveWithoutContainer(requestMessage);
 			}
 			else {
+				if (this.idleReplyContainerTimeout > 0) {
+					synchronized(this.lifeCycleMonitor) {
+						this.lastSend = System.currentTimeMillis();
+						if (!this.replyContainer.isRunning()) {
+							if (logger.isDebugEnabled()) {
+								logger.debug(this.getComponentName() + ": Starting reply container.");
+							}
+							this.replyContainer.start();
+							this.idleTask = getTaskScheduler().scheduleAtFixedRate(new IdleContainerStopper(),
+									this.idleReplyContainerTimeout / 2);
```

Comment:
Fixed rate Vs. Fixed delay is unimportant here because the task is so short-lived. They are effectively the same thing.

The `/ 2` is just to make it a little more accurate; the max stop time is 1.5 the timeout with an average of 1.25 the timeout. Without `/ 2', the max stop time is 2x the timeout with an average of 1.5x.